### PR TITLE
Move paste-related codes to qtermwidget & some fixes for pasting

### PR DESCRIPTION
**Pull request by yan12125**
Most codes are from https://github.com/lxqt/qterminal/pull/309

Fixes https://github.com/lxqt/qterminal/issues/43
Fixes https://github.com/lxqt/qterminal/issues/741

### CODE REVIEW

**Comment by yan12125** [comment]
Hi @tsujan could you have a look at this pull request and the dependent one https://github.com/lxqt/qtermwidget/pull/387? Specifially, I'd like to know if it is a good idea to remove the "Action after paste" option.

**Comment by tsujan** [comment]
Yes, of course! I might find the time tomorrow night or a little later but will definitely do it.

**Comment by tsujan** [comment]
As for "Action after paste", I can tell you right now that anything other than scrolling to the bottom feels counterintuitive to me. So, I agree with its removal.

**Comment by ito32bit** [comment]
I use "No move" of "Action after paste" option.
However, I may be the only user using this option.

**Comment by yan12125** [comment]
> I use "No move" of "Action after paste" option.

Thanks a lot for the feedback! Could you share about the need of that option?

Actually, I have "No move" before I got the bug report #741. I don't remember why I used that. Maybe because it is the default, or "Move end" is broken (#43, #741), which will be fixed after this PR.

**Review by tsujan — APPROVED** [review]
I couldn't find a problem in the code of this PR or that of https://github.com/lxqt/qtermwidget/pull/387. I also tried them after reading their corresponding bug reports and by testing various combinations of settings. In all cases, they worked as expected.

GTM.

**Comment by tsujan** [comment]
> I use "No move" of "Action after paste" option.

IMHO, the lack of scrolling down on pasting was dangerous (I always scrolled down manually before pasting) and scrolling up didn't make sense. With these PRs, the view is scrolled down when something is pasted, so that the user couldn't miss the pasted text (before hitting `Enter`).

**Comment by tsujan** [comment]
@yan12125 
This isn't directly about the PRs but only a suggestion: What about adding this tooltip to the option that disables the bracketed mode (my old PR): "Bracketed paste mode can be useful with multi-line texts."

**Comment by ito32bit** [comment]
> Thanks a lot for the feedback! Could you share about the need of that option?

For example, when I install a package, I do the following:

1. type `apt depends lxqt lxqt-core` and press Enter key (List the dependent packages)
1. type `sudo apt install `
1. scroll with mouse or Shift+PgUp or Shift+Up ...
1. [1] select a package name from "Recommends:" or "Suggests:" with the mouse
1. [2] press Shift+Insert
1. repeat [1] and [2]
1. press Shift+End to scroll to bottom
1. press Enter key to install packages

**Comment by yan12125** [comment]
> For example, when I install a package, I do the following:

Thanks a lot for sharing it! It is indeed a common scenario, so I dropped the last commit and just make "Scrolling to bottom" the default for new users.

File: src/properties.cpp [hunk 1]
```diff
@@ -111,7 +111,8 @@ void Properties::loadSettings()
     /* default to BlockCursor */
     keyboardCursorShape = m_settings->value(QLatin1String("KeyboardCursorShape"), 0).toInt();
     hideTabBarWithOneTab = m_settings->value(QLatin1String("HideTabBarWithOneTab"), false).toBool();
-    m_motionAfterPaste = m_settings->value(QLatin1String("MotionAfterPaste"), 0).toInt();
+    // For "Motion after paste", 2 (scrolling to botoom) makes more sense
```
Comment (tsujan) [inline src/properties.cpp:114]: Typo: botoom → bottom

Comment (yan12125) [reply src/properties.cpp]: Oops fixed. Thanks!

**Comment by yan12125** [comment]
> What about adding this tooltip to the option that disables the bracketed mode (my old PR): "Bracketed paste mode can be useful with multi-line texts."

Sorry forgot about this. Added in the third commit.

**Comment by yan12125** [comment]
@tsujan does the latest revision of this PR still look good?

**Comment by tsujan** [comment]
@yan12125
Yes, it's good. I don't think there's any problem but, if there is, we'll find it in practice.

**Comment by yan12125** [comment]
Thanks a lot for reviewing! I merged this and the depended PR in qtermwidget.

**Pull request merged**